rust/src/lib.rs

```diff
@@ -1,14 +1,208 @@
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
+use std::fs;
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+static DATA_FOLDER: &'static str = "src/_data";
+static CARGO_MANIFEST_DIR: &'static str = env!("CARGO_MANIFEST_DIR");
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Name {
+    pub ar: String,
+    pub en: String,
+    pub fr: String,
+}
+
+impl std::fmt::Display for Name {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let obj = json!({"ar": self.ar, "en": self.en, "fr": self.fr});
+        write!(f, "{}", serde_json::to_string_pretty(&obj).unwrap())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub enum Type {
+    #[serde(rename = "UNIVERSITY")]
+    University,
+    #[serde(rename = "ACADEMY")]
+    Academy,
+    #[serde(rename = "PRIVATE_SCHOOL")]
+    PrivateSchool,
+    #[serde(rename = "INSTITUTE")]
+    Institute,
+    #[serde(rename = "FACULTY")]
+    Faculty,
+    #[serde(rename = "DEPARTMENT")]
+    Department,
+    #[serde(rename = "SPECIALTY")]
+    Specialty,
+    #[serde(rename = "SECTOR")]
+    Sector,
+}
+
+impl std::fmt::Display for Type {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Type::University => write!(f, "UNIVERSITY"),
+            Type::Academy => write!(f, "ACADEMY"),
+            Type::PrivateSchool => write!(f, "PRIVATE_SCHOOL"),
+            Type::Institute => write!(f, "INSTITUTE"),
+            Type::Faculty => write!(f, "FACULTY"),
+            Type::Department => write!(f, "DEPARTMENT"),
+            Type::Specialty => write!(f, "SPECIALTY"),
+            Type::Sector => write!(f, "SECTOR"),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct Terms {
+    #[serde(rename = "perYear")]
+    pub per_year: usize,
+    pub slots: Vec<usize>,
+}
+
+impl std::fmt::Display for Terms {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let obj = json!({"perYear": self.per_year, "slots": self.slots});
+        write!(f, "{}", serde_json::to_string_pretty(&obj).unwrap())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Schema {
+    pub name: Name,
+    #[serde(rename = "type")]
+    pub ty: Type,
```

Comment:
I'd say the first naming is the most idiomatic way in Rust community, as you know we can't name `type` as a variable or a struct field because it is a reserved keyword in the language.

Here's some example where you can find the `ty` naming convention:
- [rust-lang](https://github.com/search?q=repo%3Arust-lang%2Frust+.ty+&type=code)
- [serde](https://github.com/search?q=repo%3Aserde-rs%2Fserde%20.ty%20&type=code)
- [rkyv](https://github.com/search?q=repo%3Arkyv%2Frkyv%20.ty%20&type=code) (btw we might use this instead of serde in the future 😅)